src/app/Notifications/NotificationCenter.tsx

```diff
@@ -37,35 +37,79 @@
+    {title: "Completed Actions", isExpanded: true, notifications: [] as Notification[], unreadCount: 0},
+    {title: "Network Info", isExpanded: false, notifications: [] as Notification[], unreadCount: 0},
+    {title: "Problems", isExpanded: false, notifications: [] as Notification[], unreadCount: 0}
+  ] as NotificationDrawerCategory[]);
+  const unreadProblemsCount = drawerCategories[PROBLEMS_CATEGORY_IDX].unreadCount;
+
+  const countUnreadNotifications = (notifications: Notification[]) => {
+    return notifications.filter(n => !n.read).length;
+  }
 
   React.useEffect(() => {
-    const sub = context.notifications().subscribe(setNotifications);
+    const sub = combineLatest([context.actionsNotifications(), context.networkInfoNotifications(), context.problemsNotifications()])
+    .subscribe(notificationLists => {
+      const updatedDrawerCategories = [...drawerCategories];
+
+      updatedDrawerCategories.map((category: NotificationDrawerCategory, idx) => {
+        category.notifications = notificationLists[idx];
+        category.unreadCount = countUnreadNotifications(notificationLists[idx]);
+      });
+
+      setDrawerCategories(updatedDrawerCategories);
+    });
     return () => sub.unsubscribe();
-  }, [context, context.notifications, notifications, setNotifications]);
+  },[context, context.notifications, setDrawerCategories]);
 
   React.useEffect(() => {
-    const sub = context.unreadNotifications().subscribe(s => setUnreadNotificationsCount(s.length));
+    const sub = context.unreadNotifications().subscribe(s => {
+      setTotalUnreadNotificationsCount(s.length)});
     return () => sub.unsubscribe();
-  }, [context, context.unreadNotifications, unreadNotificationsCount, setUnreadNotificationsCount]);
+  }, [context, context.unreadNotifications, setTotalUnreadNotificationsCount]);
+
+  const handleToggleDropdown = React.useCallback(() => {
+    setHeaderDropdownOpen(v => !v);
+  }, [setHeaderDropdownOpen]);
+
+  const handleToggleExpandCategory = React.useCallback((categoryIdx) => {
+    const updatedDrawerCategories = [...drawerCategories];
+    updatedDrawerCategories.map((category: NotificationDrawerCategory, idx) => {
+        category.isExpanded = (idx === categoryIdx) ? !category.isExpanded : false
+      });
+
+    setDrawerCategories(updatedDrawerCategories);
+  }, [setDrawerCategories]);
+
+  // Expands the Problems tab when unread errors/warnings are present
+  React.useEffect(() => {
+    if(drawerCategories[PROBLEMS_CATEGORY_IDX].unreadCount === 0) {
+      return;
+    }
 
-  const handleToggleDropdown = () => {
-    setHeaderDropdownOpen(!isHeaderDropdownOpen);
-  };
+    const updatedDrawerCategories = [...drawerCategories];
+    updatedDrawerCategories.map((category: NotificationDrawerCategory, idx) => {
+      category.isExpanded = (idx === PROBLEMS_CATEGORY_IDX) ? true : false;
+    });
+    setDrawerCategories(updatedDrawerCategories);
+  }, [setDrawerCategories, unreadProblemsCount]);
```

Comment:
Here too. Since there is also the `count === 0` guard at the top, should this hook dependency array include the `drawerCategories` state item?

Reply:
Originally I was using `unreadProblemsCount = drawerCategories[PROBLEMS_CATEGORY_IDX].unreadCount;`. I replaced this with `drawerCategories` to simplify this array.